xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -794,28 +794,60 @@ CPVRTimerInfoTagPtr CPVRTimers::GetRecordingTimerForRecording(const CPVRRecordin
     for (const auto &timersEntry : tagsEntry.second)
     {
       if (timersEntry->IsRecording() &&
-          !timersEntry->IsTimerRule() &&
+         !timersEntry->IsTimerRule() &&
           timersEntry->m_iClientId == recording.ClientID() &&
           timersEntry->m_iClientChannelUid == recording.ChannelUid())
       {
-        // first, match epg event uids, if available
-        if (timersEntry->UniqueBroadcastID() == recording.BroadcastUid() &&
-            timersEntry->UniqueBroadcastID() != EPG_TAG_INVALID_UID)
-          return timersEntry;
-
-        // alternatively, match start and end times
-        const CDateTime timerStart = timersEntry->StartAsUTC() - CDateTimeSpan(0, 0, timersEntry->m_iMarginStart, 0);
-        const CDateTime timerEnd = timersEntry->EndAsUTC() + CDateTimeSpan(0, 0, timersEntry->m_iMarginEnd, 0);
-        if (timerStart <= recording.RecordingTimeAsUTC() &&
-            timerEnd >= recording.EndTimeAsUTC())
-          return timersEntry;
+        if (recording.ChannelUid() != PVR_CHANNEL_INVALID_UID && recording.RecordingState() != PVR_RECORDED_TV_ITEM_STATE_INVALID)
+        {
+          if (recording.IsInProgress())
+            return timersEntry;
+        }
+        else
+        {
+          // first, match epg event uids, if available
+          if (timersEntry->UniqueBroadcastID() == recording.BroadcastUid() &&
+              timersEntry->UniqueBroadcastID() != EPG_TAG_INVALID_UID)
+            return timersEntry;
+
+          // alternatively, match start and end times
+          const CDateTime timerStart = timersEntry->StartAsUTC() - CDateTimeSpan(0, 0, timersEntry->m_iMarginStart, 0);
+          const CDateTime timerEnd = timersEntry->EndAsUTC() + CDateTimeSpan(0, 0, timersEntry->m_iMarginEnd, 0);
+          if (timerStart <= recording.RecordingTimeAsUTC() &&
+              timerEnd >= recording.EndTimeAsUTC())
+            return timersEntry;
+        }
       }
     }
   }
 
   return CPVRTimerInfoTagPtr();
 }
 
+/* Or with 'ChannelUid' and 'RecordingState' mandatory it becomes:
```

Comment:
We must make channel uid mandatory, but IMO tehre will always be cases where a recording was recorded on a channel is not or no longer available in Kodi. Thus, we must live with PVR_CHANNEL_INVALID_UID values...